### ISSUE TITLE
test

### ISSUE THREAD

**Issue by SteveLauC** [SteveLauC/test_topgrade_pip#1]
### Expected Behavior

[34mDEBUG[0m Configuration at /home/steve/.config/topgrade/topgrade.toml
[34mDEBUG[0m Path ~/Documents/Notes expanded to /home/steve/Documents/Notes
[34mDEBUG[0m Path ~/Documents/dotfiles expanded to /home/steve/Documents/dotfiles
[34mDEBUG[0m Loaded configuration: ConfigFile { include: None, misc: Some(Misc { pre_sudo: Some(true), git_repos: None, predefined_git_repos: None, disable: Some([Pip3, GnomeShellExtensions, Node, Rustup]), ignore_failures: None, remote_topgrades: None, remote_topgrade_path: None, ssh_arguments: None, git_arguments: None, tmux_arguments: None, set_title: None, display_time: None, assume_yes: Some(true), yay_arguments: None, aura_aur_arguments: None, aura_pacman_arguments: None, no_retry: None, run_in_tmux: None, cleanup: None, notify_each_step: None, accept_all_windows_updates: None, skip_notify: None, bashit_branch: None, only: None, no_self_update: None }), sudo_command: None, pre_commands: None, post_commands: None, commands: None, python: None, composer: None, brew: None, linux: None, git: Some(Git { max_concurrency: None, arguments: None, repos: Some(["/home/steve/Documents/Notes", "/home/steve/Documents/dotfiles"]), pull_predefined: Some(false) }), windows: None, npm: None, yarn: None, vim: None, firmware: None, vagrant: None, flatpak: None, distrobox: None }
[34mDEBUG[0m Version: 11.0.2
[34mDEBUG[0m OS: x86_64-unknown-linux-gnu
[34mDEBUG[0m Args { inner: ["topgrade", "-v", "--dry-run"] }
[34mDEBUG[0m Binary path: Ok("/home/steve/.cargo/bin/topgrade")
[34mDEBUG[0m Self Update: false
[34mDEBUG[0m Detected "/usr/bin/git" as "git"
[34mDEBUG[0m Cannot find "pwsh"
[34mDEBUG[0m Cannot find "powershell"
[34mDEBUG[0m Path "/home/steve/.config/emacs" doesn't exist
[34mDEBUG[0m Path "/home/steve/.emacs.d" doesn't exist
[34mDEBUG[0m Cannot find "doas"
[34mDEBUG[0m Cannot find "please"
[34mDEBUG[0m Detected "/usr/bin/sudo" as "sudo"
―― 19:16:59 - Sudo ――
Dry running: /usr/bin/sudo -v
[34mDEBUG[0m Step "packer.nu"
[34mDEBUG[0m Step "System update"
―― 19:16:59 - System update ――
[34mDEBUG[0m Cannot find "rpm-ostree"
[34mDEBUG[0m Detected "/usr/bin/dnf" as "dnf"
Dry running: /usr/bin/sudo /usr/bin/dnf upgrade -y
[34mDEBUG[0m Step "config-update"
[34mDEBUG[0m Step "Brew"
[34mDEBUG[0m Step "am"
[34mDEBUG[0m Step "appman"
[34mDEBUG[0m Step "deb-get"
[34mDEBUG[0m Step "toolbx"
[34mDEBUG[0m Step "Flatpak"
[34mDEBUG[0m Detected "/usr/bin/flatpak" as "flatpak"
―― 19:16:59 - Flatpak User Packages ――
Dry running: /usr/bin/flatpak update --user -y
―― 19:16:59 - Flatpak System Packages ――
Dry running: /usr/bin/flatpak update --system -y
[34mDEBUG[0m Step "snap"
[34mDEBUG[0m Step "pacstall"
[34mDEBUG[0m Step "pacdef"
[34mDEBUG[0m Step "protonup"
[34mDEBUG[0m Step "distrobox"
[34mDEBUG[0m Step "dkp-pacman"
[34mDEBUG[0m Step "pihole"
[34mDEBUG[0m Step "Firmware upgrades"
[34mDEBUG[0m Detected "/usr/bin/fwupdmgr" as "fwupdmgr"
[34mDEBUG[0m Executing command `uname -r`
[34mDEBUG[0m Uname output: 6.3.8-100.fc37.x86_64

―― 19:16:59 - Firmware upgrades ――
Dry running: /usr/bin/fwupdmgr refresh
Dry running: /usr/bin/fwupdmgr get-updates
[34mDEBUG[0m Step "Restarts"
[34mDEBUG[0m Step "yadm"
[34mDEBUG[0m Step "nix"
[34mDEBUG[0m Step "guix"
[34mDEBUG[0m Step "home-manager"
[34mDEBUG[0m Step "asdf"
[34mDEBUG[0m Step "pkgin"
[34mDEBUG[0m Step "bun"
[34mDEBUG[0m Step "zr"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Step "antibody"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Step "antidote"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Step "antigen"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Path "/home/steve/.zshrc" exists
[34mDEBUG[0m Step "zgenom"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Path "/home/steve/.zshrc" exists
[34mDEBUG[0m Step "zplug"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Path "/home/steve/.zshrc" exists
[34mDEBUG[0m Step "zinit"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Path "/home/steve/.zshrc" exists
[34mDEBUG[0m Step "zi"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Path "/home/steve/.zshrc" exists
[34mDEBUG[0m Step "zim"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Executing command `zsh -c '[[ -n ${ZIM_HOME} ]] && print -n ${ZIM_HOME}'`
[34mDEBUG[0m Command failed: Err(
   0: [91mCommand failed: `zsh -c '[[ -n ${ZIM_HOME} ]] && print -n ${ZIM_HOME}'`[0m
   1: [91m`zsh` failed: exit status: 1[0m

Location:
   [35msrc/steps/zsh.rs[0m:[35m151[0m

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  [96m                               ⋮ 1 frame hidden ⋮                               [0m
   2: [91m<E as eyre::context::ext::StdError>::ext_report[0m[90m::h042744fad1a3836b[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   3: [91mtopgrade::command::CommandExt::output_checked[0m[90m::h3c5e4c271719a100[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   4: [91mtopgrade::runner::Runner::execute[0m[90m::h6b5575decf558889[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   5: [91mtopgrade::run[0m[90m::h52b92440b4fa07c7[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   6: [91mtopgrade::main[0m[90m::h0725fd2e99cd307a[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   7: [32mstd::sys_common::backtrace::__rust_begin_short_backtrace[0m[90m::h2f44ff400a59396a[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   8: [32mmain[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   9: [91m__libc_start_call_main[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
  10: [32m__libc_start_main@GLIBC_2.2.5[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
  11: [32m_start[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.)
[34mDEBUG[0m Step "oh-my-zsh"
[34mDEBUG[0m Detected "/usr/bin/zsh" as "zsh"
[34mDEBUG[0m Path "/home/steve/.oh-my-zsh" exists
―― 19:16:59 - oh-my-zsh ――
[34mDEBUG[0m Executing command `zsh -c 'test $ZSH_CUSTOM && echo -n $ZSH_CUSTOM'`
[34mDEBUG[0m Command failed: Err(
   0: [91mCommand failed: `zsh -c 'test $ZSH_CUSTOM && echo -n $ZSH_CUSTOM'`[0m
   1: [91m`zsh` failed: exit status: 1[0m

Location:
   [35msrc/steps/zsh.rs[0m:[35m181[0m

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  [96m                               ⋮ 1 frame hidden ⋮                               [0m
   2: [91m<E as eyre::context::ext::StdError>::ext_report[0m[90m::h042744fad1a3836b[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   3: [91mtopgrade::command::CommandExt::output_checked[0m[90m::h3c5e4c271719a100[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   4: [91mtopgrade::steps::zsh::run_oh_my_zsh[0m[90m::h7d9159ce289021fc[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   5: [91mtopgrade::runner::Runner::execute[0m[90m::hc05ef8c8ee8d28f0[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   6: [91mtopgrade::run[0m[90m::h52b92440b4fa07c7[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   7: [91mtopgrade::main[0m[90m::h0725fd2e99cd307a[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   8: [32mstd::sys_common::backtrace::__rust_begin_short_backtrace[0m[90m::h2f44ff400a59396a[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
   9: [32mmain[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
  10: [91m__libc_start_call_main[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
  11: [32m__libc_start_main@GLIBC_2.2.5[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m
  12: [32m_start[0m[90m<unknown>[0m
      at [35m<unknown source file>[0m:[35m<unknown line>[0m

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.)
[34mDEBUG[0m Running zsh returned Command failed: `zsh -c 'test $ZSH_CUSTOM && echo -n $ZSH_CUSTOM'`. Using default path: /home/steve/.oh-my-zsh/custom
[34mDEBUG[0m oh-my-zsh custom dir: /home/steve/.oh-my-zsh/custom
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom/plugins is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom/plugins/example is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom/plugins/zsh-autosuggestions is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom/themes is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m /home/steve/.oh-my-zsh/custom/themes/example.zsh-theme is a file. Checking /home/steve/.oh-my-zsh/custom/themes
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom/themes is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m /home/steve/.oh-my-zsh/custom/example.zsh is a file. Checking /home/steve/.oh-my-zsh/custom
[34mDEBUG[0m Checking if /home/steve/.oh-my-zsh/custom is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
Pulling custom plugins and themes
Would pull /home/steve/.oh-my-zsh/custom/plugins/zsh-autosuggestions
Would pull /home/steve/.oh-my-zsh/custom/plugins/zsh-syntax-highlighting
Dry running: zsh /home/steve/.oh-my-zsh/tools/upgrade.sh
[34mDEBUG[0m Step "oh-my-bash"
[34mDEBUG[0m Detected "/usr/bin/bash" as "bash"
[34mDEBUG[0m Step "fisher"
[34mDEBUG[0m Step "bash-it"
[34mDEBUG[0m Step "oh-my-fish"
[34mDEBUG[0m Step "fish-plug"
[34mDEBUG[0m Step "fundle"
[34mDEBUG[0m Step "tmux"
[34mDEBUG[0m Step "TLDR"
[34mDEBUG[0m Step "pearl"
[34mDEBUG[0m Step "SDKMAN!"
[34mDEBUG[0m Detected "/usr/bin/bash" as "bash"
[34mDEBUG[0m Step "rcm"
[34mDEBUG[0m Step "maza"
[34mDEBUG[0m Step "apm"
[34mDEBUG[0m Step "fossil"
[34mDEBUG[0m Step "juliaup"
[34mDEBUG[0m Step ".NET"
[34mDEBUG[0m Step "choosenim"
[34mDEBUG[0m Step "cargo"
[34mDEBUG[0m Path "/home/steve/.cargo" exists
[34mDEBUG[0m Detected "/home/steve/.cargo/bin/cargo" as "cargo"
[34mDEBUG[0m Path "/home/steve/.cargo/.crates.toml" exists
―― 19:16:59 - Cargo ――
[34mDEBUG[0m Detected "/home/steve/.cargo/bin/cargo-install-update" as "cargo-install-update"
Dry running: /home/steve/.cargo/bin/cargo-install-update install-update --git --all
[34mDEBUG[0m Step "Flutter"
[34mDEBUG[0m Step "go-global-update"
[34mDEBUG[0m Detected "/usr/bin/go" as "go"
[34mDEBUG[0m Executing command `/usr/bin/go env GOPATH`
[34mDEBUG[0m Step "gup"
[34mDEBUG[0m Detected "/usr/bin/go" as "go"
[34mDEBUG[0m Executing command `/usr/bin/go env GOPATH`
[34mDEBUG[0m Step "Emacs"
[34mDEBUG[0m Step "opam"
[34mDEBUG[0m Step "vcpkg"
[34mDEBUG[0m Step "pipx"
[34mDEBUG[0m Step "conda"
[34mDEBUG[0m Step "mamba"
[34mDEBUG[0m Step "pip-review"
[34mDEBUG[0m Step "pip-review (local)"
[34mDEBUG[0m Step "pipupgrade"
[34mDEBUG[0m Step "ghcup"
[34mDEBUG[0m Step "stack"
[34mDEBUG[0m Step "tlmgr"
[34mDEBUG[0m Step "myrepos"
[34mDEBUG[0m Step "chezmoi"
[34mDEBUG[0m Step "jetpack"
[34mDEBUG[0m Step "vim"
[34mDEBUG[0m Step "Neovim"
[34mDEBUG[0m Detected "/home/steve/.local/share/bob/nvim-bin/nvim" as "nvim"
[34mDEBUG[0m Path "/home/steve/.config/nvim/init.lua" exists
―― 19:16:59 - Neovim ――
[34mDEBUG[0m Wrote vim script to "/tmp/.tmpHLxJka"
Dry running: /home/steve/.local/share/bob/nvim-bin/nvim -u /home/steve/.config/nvim/init.lua --headless -V1 -nS /tmp/.tmpHLxJka
[34mDEBUG[0m Step "The Ultimate vimrc"
[34mDEBUG[0m Step "voom"
[34mDEBUG[0m Step "Kakoune"
[34mDEBUG[0m Step "helix"
[34mDEBUG[0m Step "yarn"
[34mDEBUG[0m Step "pnpm"
[34mDEBUG[0m Step "Containers"
[34mDEBUG[0m Detected "/usr/bin/docker" as "docker"
[34mDEBUG[0m Using container runtime '/usr/bin/docker'
―― 19:16:59 - Containers ――
[34mDEBUG[0m Querying '/usr/bin/docker image ls --format "{{.Repository}}:{{.Tag}}/{{.ID}}"' for containers
[34mDEBUG[0m Executing command `/usr/bin/docker image ls --format '{{.Repository}}:{{.Tag}} {{.ID}}'`
[34mDEBUG[0m Containers to inspect: []
[34mDEBUG[0m Step "deno"
[34mDEBUG[0m Step "composer"
[34mDEBUG[0m Step "krew"
[34mDEBUG[0m Step "helm"
[34mDEBUG[0m Step "gem"
[34mDEBUG[0m Step "rubygems"
[34mDEBUG[0m Step "julia"
[34mDEBUG[0m Step "haxelib"
[34mDEBUG[0m Step "sheldon"
[34mDEBUG[0m Step "stew"
[34mDEBUG[0m Step "rtcl"
[34mDEBUG[0m Step "bin"
[34mDEBUG[0m Step "gcloud"
[34mDEBUG[0m Step "micro"
[34mDEBUG[0m Step "raco"
[34mDEBUG[0m Step "spicetify"
[34mDEBUG[0m Step "GitHub CLI Extensions"
[34mDEBUG[0m Checking if /home/steve/Documents/Notes is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Checking if /home/steve/Documents/dotfiles is a git repository
[34mDEBUG[0m Executing command `/usr/bin/git rev-parse --show-toplevel`
[34mDEBUG[0m Step "Git repositories"
―― 19:16:59 - Git repositories ――
Would pull /home/steve/Documents/Notes
Would pull /home/steve/Documents/dotfiles
[34mDEBUG[0m Step "Vagrant boxes"
―― 19:16:59 - Summary ――
packer.nu: SKIPPED: Cannot find "nu" in PATH
System update: OK
config-update: SKIPPED: Skipped in --yes
Brew: SKIPPED: Cannot find "brew" in PATH
am: SKIPPED: Cannot find "am" in PATH
appman: SKIPPED: Cannot find "appman" in PATH
deb-get: SKIPPED: Cannot find "deb-get" in PATH
toolbx: SKIPPED: Cannot find "toolbox" in PATH
Flatpak: OK
snap: SKIPPED: Cannot find "snap" in PATH
pacstall: SKIPPED: Cannot find "pacstall" in PATH
pacdef: SKIPPED: Cannot find "pacdef" in PATH
protonup: SKIPPED: Cannot find "protonup" in PATH
distrobox: SKIPPED: Cannot find "distrobox" in PATH
dkp-pacman: SKIPPED: Cannot find "dkp-pacman" in PATH
pihole: SKIPPED: Cannot find "pihole" in PATH
Firmware upgrades: OK
Restarts: SKIPPED: Cannot find "needrestart" in PATH
yadm: SKIPPED: Cannot find "yadm" in PATH
nix: SKIPPED: Cannot find "nix" in PATH
guix: SKIPPED: Cannot find "guix" in PATH
home-manager: SKIPPED: Cannot find "home-manager" in PATH
asdf: SKIPPED: Cannot find "asdf" in PATH
pkgin: SKIPPED: Cannot find "pkgin" in PATH
bun: SKIPPED: Cannot find "bun" in PATH
zr: SKIPPED: Cannot find "zr" in PATH
antibody: SKIPPED: Cannot find "antibody" in PATH
antidote: SKIPPED: Path "/home/steve/.antidote" doesn't exist
antigen: SKIPPED: Path "/home/steve/antigen.zsh" doesn't exist
zgenom: SKIPPED: Path "/home/steve/.zgenom" doesn't exist
zplug: SKIPPED: Path "/home/steve/.zplug" doesn't exist
zinit: SKIPPED: Path "/home/steve/.zinit" doesn't exist
zi: SKIPPED: Path "/home/steve/.zi" doesn't exist
zim: SKIPPED: Path "/home/steve/.zim" doesn't exist
oh-my-zsh: OK
oh-my-bash: SKIPPED: Path "/home/steve/.oh-my-bash" doesn't exist
fisher: SKIPPED: Cannot find "fish" in PATH
bash-it: SKIPPED: Path "/home/steve/.bash_it" doesn't exist
oh-my-fish: SKIPPED: Cannot find "fish" in PATH
fish-plug: SKIPPED: Cannot find "fish" in PATH
fundle: SKIPPED: Cannot find "fish" in PATH
tmux: SKIPPED: Path "/home/steve/.tmux/plugins/tpm/bin/update_plugins" doesn't exist
TLDR: SKIPPED: Cannot find "tldr" in PATH
pearl: SKIPPED: Cannot find "pearl" in PATH
SDKMAN!: SKIPPED: Path "/home/steve/.sdkman/bin/sdkman-init.sh" doesn't exist
rcm: SKIPPED: Cannot find "rcup" in PATH
maza: SKIPPED: Cannot find "maza" in PATH
apm: SKIPPED: Cannot find "apm" in PATH
fossil: SKIPPED: Cannot find "fossil" in PATH
juliaup: SKIPPED: Cannot find "juliaup" in PATH
.NET: SKIPPED: Cannot find "dotnet" in PATH
choosenim: SKIPPED: Cannot find "choosenim" in PATH
cargo: OK
Flutter: SKIPPED: Cannot find "flutter" in PATH
go-global-update: SKIPPED: Path "/home/steve/go/bin/go-global-update" doesn't exist
gup: SKIPPED: Path "/home/steve/go/bin/gup" doesn't exist
Emacs: SKIPPED: Cannot find "emacs" in PATH
opam: SKIPPED: Cannot find "opam" in PATH
vcpkg: SKIPPED: Cannot find "vcpkg" in PATH
pipx: SKIPPED: Cannot find "pipx" in PATH
conda: SKIPPED: Cannot find "conda" in PATH
mamba: SKIPPED: Cannot find "mamba" in PATH
pip-review: SKIPPED: Cannot find "pip-review" in PATH
pip-review (local): SKIPPED: Cannot find "pip-review" in PATH
pipupgrade: SKIPPED: Cannot find "pipupgrade" in PATH
ghcup: SKIPPED: Cannot find "ghcup" in PATH
stack: SKIPPED: Cannot find "stack" in PATH
tlmgr: SKIPPED: tlmgr must be explicity enabled in the configuration to run in Android/Linux
myrepos: SKIPPED: Cannot find "mr" in PATH
chezmoi: SKIPPED: Cannot find "chezmoi" in PATH
jetpack: SKIPPED: Cannot find "jetpack" in PATH
vim: SKIPPED: Cannot find "vim" in PATH
Neovim: OK
The Ultimate vimrc: SKIPPED: Path "/home/steve/.vim_runtime" doesn't exist
voom: SKIPPED: Cannot find "voom" in PATH
Kakoune: SKIPPED: Cannot find "kak" in PATH
helix: SKIPPED: Cannot find "helix" in PATH
yarn: SKIPPED: Cannot find "yarn" in PATH
pnpm: SKIPPED: Cannot find "pnpm" in PATH
Containers: OK
deno: SKIPPED: Cannot find "deno" in PATH
composer: SKIPPED: Cannot find "composer" in PATH
krew: SKIPPED: Cannot find "kubectl-krew" in PATH
helm: SKIPPED: Cannot find "helm" in PATH
gem: SKIPPED: Cannot find "gem" in PATH
rubygems: SKIPPED: Path "/home/steve/.gem" doesn't exist
julia: SKIPPED: Cannot find "julia" in PATH
haxelib: SKIPPED: Cannot find "haxelib" in PATH
sheldon: SKIPPED: Cannot find "sheldon" in PATH
stew: SKIPPED: Cannot find "stew" in PATH
rtcl: SKIPPED: Cannot find "rupdate" in PATH
bin: SKIPPED: Cannot find "bin" in PATH
gcloud: SKIPPED: Cannot find "gcloud" in PATH
micro: SKIPPED: Cannot find "micro" in PATH
raco: SKIPPED: Cannot find "raco" in PATH
spicetify: SKIPPED: Cannot find "spicetify" in PATH
GitHub CLI Extensions: SKIPPED: Cannot find "gh" in PATH
Git repositories: OK
Vagrant boxes: SKIPPED: Cannot find "vagrant" in PATH
[34mDEBUG[0m Desktop notification: Topgrade finished successfully
[32m INFO[0m [1mperform[0m[2m:[0m new
[32m INFO[0m [1mperform[0m[2m:[0m[1mwrite_command[0m[1m{[0m[3mcommand[0m[2m=[0mAuth(Some(External), Some([49, 48, 48, 48]))[1m}[0m[2m:[0m new
[32m INFO[0m [1mperform[0m[2m:[0m[1mwrite_command[0m[1m{[0m[3mcommand[0m[2m=[0mAuth(Some(External), Some([49, 48, 48, 48]))[1m}[0m[2m:[0m close
[32m INFO[0m [1mperform[0m[2m:[0m[1mread_command[0m[2m:[0m new
[32m INFO[0m [1mperform[0m[2m:[0m[1mread_command[0m[2m:[0m close
[32m INFO[0m [1mperform[0m[2m:[0m[1mwrite_command[0m[1m{[0m[3mcommand[0m[2m=[0mNegotiateUnixFD[1m}[0m[2m:[0m new
[32m INFO[0m [1mperform[0m[2m:[0m[1mwrite_command[0m[1m{[0m[3mcommand[0m[2m=[0mNegotiateUnixFD[1m}[0m[2m:[0m close
[32m INFO[0m [1mperform[0m[2m:[0m[1mread_command[0m[2m:[0m new
[32m INFO[0m [1mperform[0m[2m:[0m[1mread_command[0m[2m:[0m close
[32m INFO[0m [1mperform[0m[2m:[0m[1mwrite_command[0m[1m{[0m[3mcommand[0m[2m=[0mBegin[1m}[0m[2m:[0m new
[32m INFO[0m [1mperform[0m[2m:[0m[1mwrite_command[0m[1m{[0m[3mcommand[0m[2m=[0mBegin[1m}[0m[2m:[0m close
[32m INFO[0m [1mperform[0m[2m:[0m close
[32m INFO[0m [1msocket reader[0m[2m:[0m new
[32m INFO[0m [1msocket reader[0m[2m:[0m close

### Actual Behavior

ff

### Problem persists without calling from topgrade

- [X] Yes

### Steps to reproduce

test

### Topgrade version (topgrade -v)

ff

### Your Configuration file

ff

### Operating system/version

f

### Installation

f
